src/hosted-buttons/types.js

```diff
@@ -34,6 +34,11 @@ export type HostedButtonsInstance = {|
   render: (string | HTMLElement) => Promise<void>,
 |};
 
+export type HostedButtonPreferences = {|
+  buttonPreferences: $ReadOnlyArray<string>,
```

Comment:
Possible to narrow these `string` types to known preferences and sources? Could do that in follow-up implementation PR too

Reply:
I like that!

Reply:
Donezo funzo